etl/loader.py

```diff
@@ -35,37 +38,40 @@ def connect(self):
         # Get credentials from YAML
         credentials = self.get_credentials(self._config,'database_credentials')
         # create sqlalchemy engine
-        engine = sqlalchemy.create_engine("mysql+pymysql://{user}:{pw}@{host}/{db}"
-                               .format(user = credentials['db_username'],
-                                       host = credentials['db_hostname'],
-                                       pw = credentials['db_password'],
-                                       db = credentials['db_name']))
+        engine = sqlalchemy.create_engine("{dialect}://{user}{pw}{host}/{db}"
+                               .format(dialect = xstr(credentials['dialect_driver']),
+                                        user = xstr(credentials['db_username']),
+                                        host = ('@', '')[credentials['db_hostname'] is None] + xstr(credentials['db_hostname']),
+                                        pw = (':', '')[credentials['db_password'] is None] + xstr(credentials['db_password']),
+                                        db = xstr(credentials['db_name'])))
         # try to connect to database
         try:
             engine.connect()
             self._engine = engine
+            self._metadata = sqlalchemy.MetaData()
         except sqlalchemy.exc.OperationalError as except_detail:
             # print error details
             print("ERROR: {}".format(except_detail))
         return
 
-    def insert(self, dataframe, table_name, chunksize):
+    def insert(self, df_dict):
         '''
         Arguments:
-        dataframe -- pandas dataframe object
-        table_name -- table name in database
-        chunksize -- writes records in batches of a given size at a time. By default, all rows will be written at once.
-
+        df_dict -- dictionary containing transformed dataframes
         '''
-        # Insert DataFrame recrds one by one.
-        data.to_sql(attribute, con = self._engine, if_exists = 'append', chunksize = 1000)
-        return
+        for tablename, df in df_dict.items():
+            # If table don't exist, Create.
+            if not self._engine.dialect.has_table(self._engine, tablename):
+                print("Creating DB...")
```

Comment:
isn't it creating a table here? not to nitpick but just to avoid causing a heart attack if somebody sees this log in prod and thinks we are recreating the db.

Reply:
LOL!! You're right -- I fixed the printout to be less heart-attack-inducing.